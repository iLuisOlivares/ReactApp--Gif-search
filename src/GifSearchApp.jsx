import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifSearchApp = () => {

    const [categories, setCategories] = useState([]);

    // const handleAdd = () => {
    //     // setCategories([...categories, 'Bleach']);
    //     setCategories(cats => [...cats, 'Bleach']);
    // }

    return (
        <>
            <h2>Gif search App</h2>
            <AddCategory
                setCategories={setCategories}
            ></AddCategory>
            <hr />
            {/* <button onClick={handleAdd} > Add </button> */}
            <ul>
                {
                    categories.map((category) =>
                        <GifGrid
                            key={category}
                            category={category}
                        ></GifGrid>
                    )
                }
            </ul>

        </>);
};
