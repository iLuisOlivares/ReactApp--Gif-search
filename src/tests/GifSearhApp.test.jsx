import React from 'react';
import { shallow } from 'enzyme'
import { GifSearchApp } from '../GifSearchApp';


describe('Test de GifSearchApp', () => {

    test('snapshot GifSearch', () => {
        const wrapper = shallow(<GifSearchApp></GifSearchApp>)
        expect(wrapper).toMatchSnapshot();
    });


});
