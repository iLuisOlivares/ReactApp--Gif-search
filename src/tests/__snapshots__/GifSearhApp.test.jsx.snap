// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de GifSearchApp snapshot GifSearch 1`] = `
<Fragment>
  <div
    className="container-white"
  >
    <h2
      className="titulo"
    >
      Gif search App
    </h2>
    <AddCategory
      setCategories={[Function]}
    />
  </div>
  <hr />
  <ul />
</Fragment>
`;
